provisioning/roles/docker/files/docker-compose.yml

```diff
@@ -0,0 +1,171 @@
+version: "3"
+
+services:
+  elasticsearch:
+    image: elasticsearch:5.6.10
+    container_name: elasticsearch
+    restart: always
+    environment:
+      - "bootstrap.memory_lock=true"
+      - "ES_JAVA_OPTS=-Xms4g -Xmx4g"
```

Comment:
should we be starting with 4g? cc @jbeemster

Reply:
what's the reasoning behind those settings?

Reply:
we need to take the size of the machine into account now c.f. #173, Xms and Xmx should indeed be equal, c.f. https://www.elastic.co/guide/en/elasticsearch/reference/6.2/heap-size.html

Reply:
Let's decide and I'll act accordingly regarding heap size. From the page @BenFradet  referenced, it is indicated as following;

```
Set Xmx to no more than 50% of your physical RAM, to ensure that there is enough
physical RAM left for kernel file system caches
```

My only reasoning is to make ES able to stay up under high load, as much as possible. I remember us talking about publishing to `t2.large` as of this release and it has 8GB ram, hence 4GB allocation to ES.

Should it be less or is it fine, what do you think? @BenFradet @jbeemster

Reply:
I think 50% is good so that would be:

type | ES | loader | enrich | collector
| --- | --- | --- | --- | ---
t2.large | 4G | 0.5G | 0.5G | 0.5G
t2.xlarge | 8G | 1.5G | 1.5G | 1.5G
t2.2xlarge |16G | 3G | 3G | 3G